ci-operator/step-registry/cucushift/ota/preupgrade/cucushift-ota-preupgrade-commands.sh

```diff
@@ -239,6 +239,61 @@ function pre-OCP-53921(){
     return 0
 }
 
+function pre-OCP-53907(){
+    echo "Test Start: ${FUNCNAME[0]}"
+    local version
+    version="$(oc get clusterversion --no-headers | awk '{print $2}')"
+    if [ -z "${version}" ] ; then
+        echo "Fail to get cluster version!"
+        return 1
+    fi
+    x_ver=$( echo "${version}" | cut -f1 -d. )
+    y_ver=$( echo "${version}" | cut -f2 -d. )
+    y_ver=$((y_ver+1))
+    ver="${x_ver}.${y_ver}"
+    if ! oc adm upgrade channel candidate-${ver}; then
```

Comment:
I have some concern here, generally the target payload would be provided by ci config level, while in this testing, it depends on official candidate channel, is there a unmatched issue?

Reply:
Yes, possible.  The ci config is choosing the latest one, and the upgrade is using --tolatest, it should be the same.  There might be some edge case caused they are unmatched. But even it's unmatched, the upgrade path is also valid for this case's test.

Reply:
Once it is unmatched, the cucushift-upgrade-toimage step will fail, because there are several guard to check the target version, while the target version is from ci config.

Reply:
good catch, let me update to do post check with correct target version when unmatched.

Reply:
I am thinking if we should create a dedicated ci job (only for OTA testing) to run such testing to avoid bring some noise when running together with regular ci.